File: codechef/RRCopy.java
import java.io.*;

class Main
{
	public static void main(String[] args) throws IOException {
		BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
		int testcases= Integer.parseInt(reader.readLine());

		while(testcases-- != 0)
		{
			long N = Long.parseLong(reader.readLine());
			long count = 0;
			boolean present[] = new boolean[100000];
			String[] s = reader.readLine().split(" ");
			for(long i = 0; N-- != 0; i++)
			{
				long temp = Long.parseLong(s[(int)i]);
				if(!present[(int)temp])
				{
					present[(int)temp] = true;
					count++;
				}
			}
			System.out.println(count);
		}

	}
}

File: codechef/rrcopy.cpp
#include "cstdio"
#include "vector"

using namespace std;

typedef unsigned long long int ulint;

int main(int argc, char const *argv[])
{
	short int testCases = 0;
	
	ulint N=0;
	
	scanf("%u", &testCases);

	while(testCases--)
	{
		// for (int i = 0; i < 10000; ++i)
		// {
		// 	present[i] = false;
		// }
		vector<bool> present;
		ulint count = 0, temp=0;
		
		scanf("%llu", &N);
		while(N--)
		{
			scanf("%llu", &temp);
			temp--;
			if(present[temp] == false)
			{
				present[temp] = true;
				count++;
			}
		}
		printf("%llu\n", count);	
	}

	return 0;
}